core/model/modx/processors/context/remove.class.php

```diff
@@ -23,14 +23,31 @@ public function beforeRemove() {
     }
 
     public function afterRemove() {
+        /* Retrieve all resources from this context. */
+        $resources = $this->modx->getCollection('modResource',array(
```

Comment:
Maybe here make sense to use `getIterator` againt `getCollection `. It'll allow avoiding memory leak in a case when context had a lot of resources.